### PR TITLE
Upgrade code to be compatible with most recent Dart/Flutter

### PR DESCRIPTION
**Pull request by noordawod**
* Remove unnecessary assertions (that tested for nullability).
* Fix null safety with recent Dart version.
* Fix warnings reported by the linter.
* Use an equal sign to define default parameter values (instead of a colon).
* Use other calls to replace deprecated ones.

### CODE REVIEW

File: lib/src/http.dart [hunk 1]
```diff
@@ -205,7 +204,6 @@ class Cookies {
   }
 
   List<Cookie>? _cookies;
-  Map<String, List<Cookies>>? _sessions;
```
Comment (noordawod) [inline lib/src/http.dart:208]: Unused in code.

File: lib/src/provider/flutter_advanced_networkimage.dart [hunk 2]
```diff
@@ -18,15 +18,15 @@ typedef Future<Uint8List?> _ImageProcessing(Uint8List data);
 class AdvancedNetworkImage extends ImageProvider<AdvancedNetworkImage> {
   AdvancedNetworkImage(
     this.url, {
-    this.scale: 1.0,
+    this.scale = 1.0,
     this.width,
     this.height,
     this.header,
-    this.useDiskCache: false,
```
Comment (noordawod) [inline lib/src/provider/flutter_advanced_networkimage.dart:25]: This is wrong, the correct style for default parameters is by using an equal sign. This is fixed in all places now.

File: lib/src/http.dart [hunk 3]
```diff
@@ -149,17 +147,18 @@ Future<_Response> _sendRequest(
...
+
+    HttpClientResponse res = await (typed as FutureOr<HttpClientResponse>);
     var headers = <String, String>{};
     res.headers.forEach((key, values) {
       headers[key] = values.join(',');
     });
 
     return _Response(
       http.StreamedResponse(
-        DelegatingStream.typed<List<int>>(res).handleError(
```
Comment (noordawod) [inline lib/src/http.dart:162]: `DelegatingStream.typed()` is deprecated.

File: test/cache_test.dart [hunk 4]
```diff
@@ -4,10 +4,9 @@ import 'dart:typed_data';
 
 import 'package:path/path.dart';
 import 'package:test/test.dart';
-import 'package:flutter/services.dart';
 import 'package:path_provider/path_provider.dart';
 import 'package:flutter_test/flutter_test.dart'
-    show TestWidgetsFlutterBinding, WidgetTester, throwsAssertionError;
```
Comment (noordawod) [inline test/cache_test.dart:10]: Unused.

File: test/cache_test.dart [hunk 5]
```diff
@@ -4,10 +4,9 @@ import 'dart:typed_data';
 
 import 'package:path/path.dart';
 import 'package:test/test.dart';
-import 'package:flutter/services.dart';
```
Comment (noordawod) [inline test/cache_test.dart:7]: Unused.

File: lib/src/http.dart [hunk 6]
```diff
@@ -149,17 +147,18 @@ Future<_Response> _sendRequest(
     if (cookies != null && cookies.length > 0)
       ioRequest.cookies.addAll(cookies);
 
-    HttpClientResponse res =
-        await (stream.pipe(DelegatingStreamConsumer.typed(ioRequest))
```
Comment (noordawod) [inline lib/src/http.dart:153]: `DelegatingStreamConsumer.typed` is deprecated.

**Comment by noordawod** [comment]
@insinfo could you take a look this week please? it's quick stuff :smile:

**Comment by insinfo** [comment]
@noordawod 
Thank you very much for the PR, I already merged and published the update

**Pull request merged**